scripts/dockerrun.sh

```diff
@@ -87,8 +87,43 @@ fi
 #  VSO
 [ ! -z "$BUILD_BUILDID" ] && DOTNET_BUILD_CONTAINER_NAME="${BUILD_BUILDID}-${BUILD_BUILDNUMBER}"
 
+# Executes a command and retries if it fails.
+# NOTE: This function is the exact copy from init-docker.sh.
+# Reason for not invoking init.docker.sh directly is since that script 
+# also performs cleanup, which we do not want in this case.
```

Comment:
@MichaelSimons That's true, they aren't normally restored until midway into the build, inside the container. I think this comment should read something like this, then:

> Reason for not invoking init-docker.sh directly is that it is not readily available at this point in the build.

(Maybe it'd even link to a core-setup issue tracking that.)

I agree that checked-in-script auto-update makes a lot of sense here. I think we could probably get that going before we get BuildTools segmented in a way that works here.